lib/lte_link_control/Kconfig

```diff
@@ -365,6 +368,12 @@ config LTE_LC_TRACE
 	  what has happened and has no associated payload.
 	  It is intended to be used for debugging purposes.
 
+config LTE_EXTENDED_FEATURES
```

Comment:
Cannot add comment to modem shell code so adding here.
This Kconfig should be enabled/selected in `samples/cellular/modem_shell/Kconfig` when `CONFIG_MOSH_LINK` is enabled. `CONFIG_LTE_EXTENDED_FEATURES` is enabled in MOSH by default but if you compile it with `CONFIG_MOSH_LOCATION=n`, extended features are disabled.

The places where we might need this flag might be a little harder to spot easily or all lte_lc functions should be checked. E.g., I found `subsys/caf/modules/net_state.c` using `LTE_LC_EVT_EDRX_UPDATE` event.

I'm trying to think if there are other libs/samples that could need similar change. I'm sure these are not the only ones.

Reply:
To avoid this kind of hard to spot problems, why not remove the code returning `-EOPNOTSUPP` when an unsupported function is called? This would cause a build time failure when trying to use "extended features" with the relevant option disabled.

We could even consider flagging affected event IDs in `lte_lc.h` to make sure the application isn't expecting an event which will never come, if events are affected.